src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -611,6 +614,8 @@ protected override void Dispose(bool disposing)
                         base.Dispose(disposing);
                     }
                 }  // finally
+
+                s_byteBufferPool.Free(_buffer);
```

Comment:
I am not familiar with the deflater codebase, but quite familiar with this pool. One thing to watch out when using ObjectPool, especially if it is static, is to not return back large objects. 
If _buffer may expand, make sure you are not returning really big ones.